### ISSUE TITLE
Data Attributes from JavaScript

### ISSUE THREAD

**Issue by mihawk90** [Prinzhorn/skrollr#718]
TL;DR; is under the horizontal line at the end :stuck_out_tongue: 
Heya,

found this project a few days ago and looks really good to me.
Now in a new project I'm working on I'm going to use this, but I didn't like having to put the data attributes into my html statically, I needed a (somewhat) dynamic approach (aka external style sheet).

I saw the skrollr-stylesheet plugin, but I didn't like the approach of a none-valid CSS sheet in my site (that standpoint is probably debatable but that's just me), so I needed something different.

I'm now using a Javascript approach to apply the skrollr-data tags, the file looks like this (based on [this article](https://ihatetomatoes.net/simple-parallax-scrolling-tutorial/)):

```javascript
    var data = {
            "#slide-1 .bcg": {
                "start-top":    "background-position: 40% 0px;",
                "center":       "background-position: 50% 0px;",
                "top-bottom":   "background-position: 50% -100px;",
                "anchor-target": "#slide-1"
            },
            "#slide-1 .bcg .hsContainer": {
                "center":       "bottom: 200px; opacity: 1;",
                "top":          "bottom: 1200px; opacity: 0;",
                "anchor-target": "#slide-1 h2"
            },
            "#slide-2 .bcg": {
                "center":       "background-position: 50% 0px;",
                "top-bottom":   "background-position: 25% -100px;",
                "bottom-top":   "background-position: 75% 100px;",
                "anchor-target": "#slide-2"
            },
            "#slide-2 .bcg .hsContainer": {
                "center":       "opacity: 1",
                "center-top":   "opacity: 0",
                "-100-bottom":  "opacity: 0;",
                "anchor-target": "#slide-2"
            }
```

It's a little longer, but you get the point.

I then needed to parse the object and apply the data attributes to the file:
```javascript
    // http://stackoverflow.com/a/921808
    for (element in data) {
        if (data.hasOwnProperty(element)) {
            for (dataProp in data[element]) {
                // important check that this is objects own property 
                // not from prototype prop inherited
                if (data[element].hasOwnProperty(dataProp)) {
                    document.querySelector(element).setAttribute("data-" + dataProp, data[element][dataProp]);
                }
            }
        }
    }
```

Initially I used a DOMContentLoaded event to apply this, then I saw the beforeRender option property and used that.

I'm pretty sure I'm not the first one to get this idea, but I couldn't find anything about it anywhere. Suggestions are welcome of course.

---

Anyway, my point is it would be nice if skrollr had a similar object parsing function integrated (or does it already and I'm blind?), so one just needed to pass the object/function and you're done.

Plus points:
* Content and styling separated
* data attributes separated from template, so integratable into CMS if needed
* No Invalid CSS sheet (skrollr-stylesheets)
* object notation is somewhat similar to CSS notation - design-type-guy could easily adapt
* no unused data-attributes in case of no-JS (whoever does this these days, you never know)
* most likely faster then parsing a stylesheet/text like skrollr-stylesheets does

Minus Points
* longer execution time, maybe? I have no benchmarks or anything
* ???

**Comment by Prinzhorn** [comment]
> Initially I used a DOMContentLoaded event to apply this, then I saw the beforeRender option property and used that.

Don't, it's called 60 times per second while scrolling.

I've only skimmed through your post and yes, you're not the first one. But you can easily write a wrapper around the `refresh` method and call it a plugin. This gives you the same functionality you want without touching skrollr at all. I don't have any plans to make major changes to skrollr and this would be one.
